tests/app/celery/test_broadcast_message_tasks.py

```diff
@@ -16,18 +22,48 @@
 from tests.conftest import set_config
 
 
-def test_send_broadcast_event_queues_up_for_active_providers(mocker, notify_api):
+def test_send_broadcast_event_queues_up_for_active_providers(mocker, notify_api, sample_service):
+    template = create_template(sample_service, BROADCAST_TYPE)
+    broadcast_message = create_broadcast_message(template, status=BroadcastStatusType.BROADCASTING)
+    event = create_broadcast_event(broadcast_message)
+
     mock_send_broadcast_provider_message = mocker.patch(
         'app.celery.broadcast_message_tasks.send_broadcast_provider_message',
     )
 
-    event_id = uuid.uuid4()
     with set_config(notify_api, 'ENABLED_CBCS', ['ee', 'vodafone']):
-        send_broadcast_event(event_id)
+        send_broadcast_event(event.id)
+
+    assert mock_send_broadcast_provider_message.apply_async.call_args_list == [
+        call(kwargs={'broadcast_event_id': event.id, 'provider': 'ee'}, queue='notify-internal-tasks'),
+        call(kwargs={'broadcast_event_id': event.id, 'provider': 'vodafone'}, queue='notify-internal-tasks')
+    ]
+
+
+def test_send_broadcast_event_only_sends_to_one_provider_if_set_on_service(
+    mocker,
+    notify_db_session,
+    notify_api,
+    sample_service
+):
+    notify_db_session.add(ServiceBroadcastProviderRestriction(
+        service=sample_service,
+        provider='vodafone'
+    ))
+
+    template = create_template(sample_service, BROADCAST_TYPE)
+    broadcast_message = create_broadcast_message(template, status=BroadcastStatusType.BROADCASTING)
+    event = create_broadcast_event(broadcast_message)
+
+    mock_send_broadcast_provider_message = mocker.patch(
+        'app.celery.broadcast_message_tasks.send_broadcast_provider_message',
+    )
+
+    with set_config(notify_api, 'ENABLED_CBCS', ['ee', 'vodafone']):
```

Comment:
Might be an edge case but wonder if it's worth testing for the case where the single restricted broadcast provider is not in `ENABLED_CBCS`